src/vs/editor/common/controller/cursor.ts

```diff
@@ -1234,11 +1234,17 @@ export class Cursor extends EventEmitter {
 	}
 
 	private _moveToBeginningOfLine(inSelectionMode: boolean, ctx: IMultipleCursorOperationContext): boolean {
-		return this._invokeForAll(ctx, (cursorIndex: number, oneCursor: OneCursor, oneCtx: IOneCursorOperationContext) => OneCursorOp.moveToBeginningOfLine(oneCursor, inSelectionMode, oneCtx));
```

Comment:
After taking a look at the code, the method ```_cursorMove``` which I used instead of ```_invokeForAll``` uses  ```_invokeForAll``` internally so I'm guessing it's not a problem?
However, I think we do need to clarify how ```Home``` and ```End``` command should behave in multi cursor mode. I'll raise this question in the related issue directly.